compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -899,10 +1006,70 @@ private STNode parseFuncDefinition(STNode metadata, boolean isObjectMember, STNo
      */
     private STNode parseFuncDefOrFuncTypeDesc(ParserRuleContext context, STNode metadata, List<STNode> qualifiers,
                                               boolean isObjectMember, boolean isObjectTypeDesc) {
-        STNode qualifierList = parseFunctionQualifiers(context, qualifiers);
+        STNode qualifierList;
+        // TODO: remove validation at this level when syncing to service-typing branch
+        if (context == ParserRuleContext.TOP_LEVEL_FUNC_DEF_OR_FUNC_TYPE_DESC) {
+            qualifierList = createFuncDefNodeList(qualifiers);
+        } else {
+            qualifierList = createMethodQualNodeList(qualifiers);
+        }
         return parseFuncDefOrFuncTypeDesc(metadata, qualifierList, isObjectMember, isObjectTypeDesc);
     }
 
+    private STNode createFuncDefNodeList(List<STNode> qualifierList) {
+        // This method is temporary. Will be removed when syncing to service-typing branch
+        // Validate qualifiers and create a STNodeList
+        for (int i = 0; i < qualifierList.size();) {
+            STNode qualifier = qualifierList.get(i);
+            if (qualifier.kind != SyntaxKind.ISOLATED_KEYWORD &&
+                    qualifier.kind != SyntaxKind.TRANSACTIONAL_KEYWORD &&
+                    qualifier.kind != SyntaxKind.PUBLIC_KEYWORD) {
+                qualifierList.remove(i);
+                if (qualifierList.size() == i) {
+                    addInvalidNodeToNextToken(qualifier, DiagnosticErrorCode.ERROR_QUALIFIER_NOT_ALLOWED,
+                            ((STToken) qualifier).text());
+                } else {
+                    STNode nextQual = qualifierList.remove(i);
+                    nextQual = SyntaxErrors.cloneWithLeadingInvalidNodeMinutiae(nextQual, qualifier,
+                            DiagnosticErrorCode.ERROR_QUALIFIER_NOT_ALLOWED, ((STToken) qualifier).text());
+                    qualifierList.add(nextQual);
+                }
+                continue;
+            }
+            i++;
+        }
+
+        return STNodeFactory.createNodeList(qualifierList);
+    }
+
+    private STNode createMethodQualNodeList(List<STNode> qualifierList) {
+        // This method is temporary. Will be removed when syncing to service-typing branch
+        // Validate qualifiers and create a STNodeList
+        for (int i = 0; i < qualifierList.size();) {
+            STNode qualifier = qualifierList.get(i);
+            if (qualifier.kind != SyntaxKind.ISOLATED_KEYWORD &&
+                    qualifier.kind != SyntaxKind.TRANSACTIONAL_KEYWORD &&
+                    qualifier.kind != SyntaxKind.REMOTE_KEYWORD &&
+                    qualifier.kind != SyntaxKind.PUBLIC_KEYWORD &&
+                    qualifier.kind != SyntaxKind.PRIVATE_KEYWORD) {
```

Comment:
Better to extract this logic inside the 'if-condition' to a separate method, something like `isValidMethodQualifier`